azurelinuxagent/ga/exthandlers.py

```diff
@@ -743,54 +745,84 @@ def report_event(self, message="", is_success=True, duration=0, log_event=True):
         add_event(name=self.ext_handler.name, version=ext_handler_version, message=message,
                   op=self.operation, is_success=is_success, duration=duration, log_event=log_event)
 
+    def _download_extension_package(self, source_uri, target_file):
+        self.logger.info("Downloading extension package: {0}", source_uri)
+        try:
+            if not self.protocol.download_ext_handler_pkg(source_uri, target_file):
+                raise Exception("Failed to download extension package - no error information is available")
+        except Exception as exception:
+            self.logger.info("Error downloading extension package: {0}", ustr(exception))
+            if os.path.exists(target_file):
+                os.remove(target_file)
+            return False
+        return True
+
+    def _unzip_extension_package(self, source_file, target_directory):
```

Comment:
Just a clarification - Would the issue of corrupt zip arise for Agent as well? and we need similar logic in `azurelinuxagent.ga.update.GuestAgent#_unpack`?

Reply:
Yup, and I intend to fix that as well. However, this is a hotfix release, so I do not want to add any other changes to it. Thanks for pointing it out.